### DISCUSSION TITLE
color_sensor.reflection() does not display the correct value

### DISCUSSION THREAD

**Discussion by ghost** [pybricks/support#1781] [Q&A]
I use two Mindstorms hubs. One with Pybricks, the other with Mindstorms, both with word blocks.
Furthermore, I use the Spike Prime Color Sensor on a WRO playing field mat.
The hub with Mindstorms displays on a white field a reflection value of about 99.
The hub with Pybricks display a reflection value of under 60.
How could this be explained?

**Comment by laurensvalk** [comment]
We scale the value a bit differently in Pybricks such that high values don't get cut off by as much.

If it already gives you 99% for your condition with the stock firmware, then putting it any closer to the floor will just saturate it at 100%, so you lose a lot of information. We wanted to make sure that it works well across a wider range of conditions.

**Comment by ghost** [comment]
I think I understand your arguments. 
The measurements were based on a model with a distance of 2 LE from matte to color sensor. Many building instructions, including those from Lego, have this distance.
I reduced the distance from 2 LEs to 1 LE and I got very usable values. 
I have presented my measurements in the attachment
![Messungen CS](https://github.com/user-attachments/assets/6ac8bdaa-0083-409b-901e-c7af06dccf9c)
.

**Reply by laurensvalk** [reply]
Very nice, thanks for sharing!

And indeed you have confirmed that with the LEGO app, it already saturates at 99% at 2 LEGO units, so putting it closer starts to reduce the resolution/contrast with the black line. This likely also means that when you put it at 1 LEGO unit from the line, it is less accurate at finding the line, because even some grey tones (i.e. partially on the line) will be reported as 99%.

Whereas with Pybricks, you still get the full resolution in that case, so grey tones are not ignored.

The aforementioned applies to the reflection command. You can also calibrate the color measurements if you like so that it matches the distance on your robot. See [this video](https://www.youtube.com/watch?v=UpA2erJfGQY) for a guide. This shows it for trains, but it works the same on a robot competition map.
